blueprints/data-solutions/composer-2/main.tf

```diff
@@ -0,0 +1,159 @@
+      module.comp-sa.iam_email
+    ]
+    "roles/composer.ServiceAgentV2Ext" = [
+      "serviceAccount:${module.project.service_accounts.robots.composer}"
+    ]
+  }
+
+  _shared_vpc_bindings = {
+    "roles/compute.networkUser" = [
+      "prj-cloudservices", "prj-robot-gke"
+    ]
+    "roles/composer.sharedVpcAgent" = [
+      "prj-robot-cs"
+    ]
+    "roles/container.hostServiceAgentUser" = [
+      "prj-robot-gke"
+    ]
+  }
+  shared_vpc_role_members = {
+    prj-cloudservices = "serviceAccount:${module.project.service_accounts.cloud_services}"
+    prj-robot-gke     = "serviceAccount:${module.project.service_accounts.robots.container-engine}"
+    prj-robot-cs      = "serviceAccount:${module.project.service_accounts.robots.composer}"
+  }
+  # reassemble in a format suitable for for_each
+  shared_vpc_bindings_map = {
+    for binding in flatten([
+      for role, members in local._shared_vpc_bindings : [
+        for member in members : { role = role, member = member }
+      ]
+    ]) : "${binding.role}-${binding.member}" => binding
+  }
+
+  shared_vpc_project = try(var.network_config.host_project, null)
+  use_shared_vpc     = var.network_config != null
+
+  vpc_self_link = (
+    local.use_shared_vpc
+    ? var.network_config.network_self_link
+    : module.vpc.0.self_link
+  )
+
+  orch_subnet = (
+    local.use_shared_vpc
+    ? var.network_config.subnet_self_link
+    : values(module.vpc.0.subnet_self_links)[0]
+  )
+
+  orch_vpc = (
+    local.use_shared_vpc
+    ? var.network_config.network_self_link
+    : module.vpc.0.self_link
+  )
+  groups = {
+    for k, v in var.groups : k => "${v}@${var.organization_domain}"
+  }
+  groups_iam = {
+    for k, v in local.groups : k => "group:${v}"
+  }
+}
+
+module "project" {
+  source          = "../../../modules/project"
+  name            = var.project_id
+  parent          = try(var.project_create.parent, null)
+  billing_account = try(var.project_create.billing_account_id, null)
+  project_create  = var.project_create != null
+  prefix          = var.project_create == null ? null : var.prefix
+  iam             = var.project_create != null ? local.iam : {}
+  iam_additive    = var.project_create == null ? local.iam : {}
+  services = [
+    "cloudkms.googleapis.com",
+    "container.googleapis.com",
+    "containerregistry.googleapis.com",
+    "composer.googleapis.com",
+    "compute.googleapis.com",
+    "iap.googleapis.com",
+    "logging.googleapis.com",
+    "monitoring.googleapis.com",
+    "networkmanagement.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "storage.googleapis.com",
+    "storage-component.googleapis.com",
+  ]
+
+  shared_vpc_service_config = local.shared_vpc_project == null ? null : {
+    attach               = true
+    host_project         = local.shared_vpc_project
+    service_identity_iam = {}
+  }
+
+  service_encryption_key_ids = {
+    composer = [try(lookup(var.service_encryption_keys, var.region, null), null)]
```

Comment:
`try` or `lookup` but I I think you don't need both

Reply:
right, removed the 'lookup'